src/core/tsi/ssl_transport_security.cc

```diff
@@ -1762,6 +1762,13 @@ void tsi_ssl_client_handshaker_factory_unref(
   tsi_ssl_handshaker_factory_unref(&factory->base);
 }
 
+tsi_ssl_client_handshaker_factory* tsi_ssl_client_handshaker_factory_ref(
+    tsi_ssl_client_handshaker_factory* client_factory) {
+  if (client_factory == nullptr) return nullptr;
+  return reinterpret_cast<tsi_ssl_client_handshaker_factory*>(
```

Comment:
To make this code a bit more readable, could we instead do:
```
if (client_factory == nullptr) return nullptr;
tsi_ssl_handshaker_factory_ref(&client_factory->base);
return client_factory
```

?

Reply:
Friendly ping :)

Reply:
I'm not 100% on this - the semantics of the `tsi_ssl_handshaker_factory_ref` that this calls is to return the factory, but's it's also just the same one that's in. I figured semantically since it returns it, there was a reason for that and it should be used?

Reply:
https://github.com/grpc/grpc/blob/1986007e1ee735ec3f977f31cc8569c5ad954c7e/src/core/tsi/ssl_transport_security.cc#L1132-L1137

Reply:
Up to you. I think they're equivalent, but I'm also totally fine leaving it as-is.

Reply:
Let's leave as-is